apps/experiments/views/experiment.py

```diff
@@ -724,26 +739,32 @@ def experiment_invitations(request, team_slug: str, experiment_id: str):
     if request.method == "POST":
         post_form = ExperimentInvitationForm(request.POST)
         if post_form.is_valid():
-            participant = Participant.objects.get_or_create(
-                team=request.team, identifier=post_form.cleaned_data["email"]
-            )[0]
             if ExperimentSession.objects.filter(
                 team=request.team,
                 experiment=experiment,
-                participant=participant,
                 status__in=["setup", "pending"],
+                participant__external_chat_id=post_form.cleaned_data["email"],
             ).exists():
-                messages.info(request, f"{participant} already has a pending invitation.")
+                participant_email = post_form.cleaned_data["email"]
+                messages.info(request, f"{participant_email} already has a pending invitation.")
             else:
-                channel = _ensure_experiment_channel_exists(experiment, platform="web", name=f"{experiment.id}-web")
-                session = ExperimentSession.objects.create(
-                    team=request.team,
-                    experiment=experiment,
-                    llm=experiment.llm,
-                    status="setup",
-                    participant=participant,
-                    experiment_channel=channel,
-                )
+                with transaction.atomic():
+                    channel = _ensure_experiment_channel_exists(experiment, platform="web", name=f"{experiment.id}-web")
+                    # TODO: Use _start_experiment_session and pass in specific kwargs
+                    session = ExperimentSession.objects.create(
+                        team=request.team,
+                        experiment=experiment,
+                        llm=experiment.llm,
+                        status="setup",
+                        experiment_channel=channel,
+                    )
+                    participant, _ = Participant.objects.get_or_create(
```

Comment:
move this above the session create so that you don't have to re-save the session after

Reply:
https://github.com/dimagi/open-chat-studio/pull/363/commits/e9032d56272b5862ba32d30d73a0adb459ed138d

Reply:
The test: https://github.com/dimagi/open-chat-studio/pull/363/commits/0e77e1442bc31d120eaa5559527538e2084eb40e